app/controllers/publish/courses/funding_type_controller.rb

```diff
@@ -92,6 +117,28 @@ def course_page_path
       def section_key
         'Funding type'
       end
+
+      def handle_redirect
+        if goto_visa_path?
+          redirect_to visa_sponsorship_path
+        elsif previous_tda_course_path?
+          redirect_to previously_defaulted_attributes
+        else
+          redirect_to next_step
+        end
+      end
+
+      def goto_visa_path?
+        params[:goto_visa] == 'true' && params[:course][:previous_tda_course] != 'true'
+      end
+
+      def previous_tda_course_path?
+        params[:course][:previous_tda_course] == 'true'
```

Comment:
Same as above

Reply:
I believe so. 

Changed to dig: https://github.com/DFE-Digital/publish-teacher-training/pull/4223/commits/9f2cda8d629b557bec7187582435e9cd917b3f5e